src/app/project/ontology/ontology.service.ts

```diff
@@ -79,4 +95,92 @@ export class OntologyService {
             return Cardinality._0_1;
         }
     }
+
+    getSuperProperty(property: ResourcePropertyDefinitionWithAllLanguages): string {
+        // get ontology from property info
+        const ontoIri = property.id.split(Constants.HashDelimiter)[0];
+
+        let superPropIri: string;
+
+        // get iri from sub properties
+        if (property.subPropertyOf.length) {
+            for (const subProp of property.subPropertyOf) {
+                const baseOntoIri = subProp.split(Constants.HashDelimiter)[0];
+                // compare with knora base ontology
+                if (baseOntoIri !== Constants.KnoraApiV2) {
+                    // the property is not a subproperty of knora base ontology
+                    // get property iri from another ontology
+                    this._cache.get('currentProjectOntologies').subscribe(
+                        (ontologies: ReadOntology[]) => {
+                            const onto = ontologies.find(i => i.id === baseOntoIri);
+                            superPropIri = onto.properties[subProp].subPropertyOf[0];
+                        }
+                    );
+                }
+
+                if (superPropIri) {
```

Comment:
Why does this if statement need to be here?

Reply:
As soon as the superPropIri is defined (or found), we can exit the for-loop.

Reply:
Maybe it's not needed. I think I found this solution somewhere on stackoverflow 🙈

Reply:
I think it would work if you put the `break` on line 117, no?

Reply:
You mean like this:
<img width="699" alt="Screen Shot 2022-02-01 at 10 10 38" src="https://user-images.githubusercontent.com/4253580/151941018-e18564c5-0ae5-40f5-9fce-6d76a05eaa29.png">

Then I got an error

Reply:
Yeah that's what I meant, good to know it doesn't work like I thought it would 😅